custom_components/auto_areas/auto_lights.py

```diff
@@ -104,30 +113,41 @@ async def initialize(self) -> None:
             self.handle_presence_state_change,
         )
 
-    async def handle_presence_state_change(
-        self, entity_id, from_state: State, to_state: State
-    ):
-        previous_state = from_state.state if from_state else ""
-        current_state = to_state.state
-
-        if previous_state == current_state:
-            return
-
-        _LOGGER.info(
-            "State change: presence entity (%s) %s -> %s",
-            entity_id,
-            previous_state,
-            current_state,
-        )
-
-        if current_state == STATE_ON:
-            if self.sleep_mode_enabled:
+    async def handle_presence_state_on(self):
+        if self.sleep_mode_enabled:
+            _LOGGER.info(
+                "Presence detected, but sleep mode is on (%s). Not turning on lights",
+                self.area_name,
+            )
+            if self.sleeping_scene_entity_id:
                 _LOGGER.info(
-                    "Sleep mode is on (%s). Not turning on lights", self.area_name
+                    "Activating sleep scene (%s) %s",
+                    self.area_name,
+                    self.sleeping_scene_entity_id,
+                )
+                await self.hass.services.async_call(
```

Comment:
@rsnodgrass While writing some BDD scenarios I was wondering if this is really what we want: the configured sleep scene would be activated every time there is motion detected while sleep mode is enabled. 

I would rather only enable the sleep scene once: at the moment when the sleep mode switch is turned on. I was thinking this could really be useful to turn down all the blinds and turn off bedroom lights when going to bed.

I am wondering: is the current behaviour correct? And if so what is your intended behaviour with regard to sleep mode scenes?